pkg/statistics/handle/storage/read.go

```diff
@@ -763,7 +783,7 @@ func loadNeededColumnHistograms(sctx sessionctx.Context, statsHandle statstypes.
 		)
 		return nil
 	}
-	statsTbl = statsTbl.Copy()
+	statsTbl = statsTbl.CopyForColumnUpdate()
```

Comment:
It can be extremely dangerous. In Golang, if you perform concurrent read and write operations on a map, it may cause a panic.

Reply:
https://pkg.go.dev/runtime#SetFinalizer

Reply:
https://pkg.go.dev/runtime#AddCleanup

Reply:
discussed offline. The map should be safe since it's copied before the update. Should be no concurrent access.

AddCleanup is potentially a good tool but out of the scope of this PR, will explore to use it in another PR

Reply:
https://github.com/pingcap/tidb/issues/63184